src/mono/AppleAppBuilder/AppleAppBuilder.cs

```diff
@@ -0,0 +1,164 @@
+    /// </summary>
+    [Required]
+    public string ProjectName { get; set; } = ""!;
+
+    /// <summary>
+    /// Target directory with *dll and other content to be AOT'd and/or bundled
+    /// </summary>
+    [Required]
+    public string AppDir { get; set; } = ""!;
+
+    /// <summary>
+    /// Path to Mono public headers (*.h)
+    /// </summary>
+    [Required]
+    public string MonoInclude { get; set; } = ""!;
+
+    /// <summary>
+    /// This library will be used as an entry-point (e.g. TestRunner.dll)
+    /// </summary>
+    [Required]
+    public string MainLibraryFileName { get; set; } = ""!;
+
+    /// <summary>
+    /// Path to a custom main.m with custom UI
+    /// A default one is used if it's not set
+    /// </summary>
+    public string? NativeMainSource { get; set; }
+
+    /// <summary>
+    /// Produce optimized binaries (e.g. use -O2 in AOT)
+    /// </summary>
+    public bool Optimized { get; set; }
+
+    /// <summary>
+    /// Target arch, can be "arm64" (device) or "x64" (simulator) at the moment
+    /// </summary>
+    [Required]
+    public string Arch { get; set; } = ""!;
+
+    /// <summary>
+    /// DEVELOPER_TEAM provisioning, needed for arm64 builds.
+    /// </summary>
+    public string? DevTeamProvisioning { get; set; }
+
+    /// <summary>
+    /// Build *.app bundle (using XCode for now)
+    /// </summary>
+    public bool BuildAppBundle { get; set; }
+
+    /// <summary>
+    /// Generate xcode project
+    /// </summary>
+    public bool GenerateXcodeProject { get; set; }
+
+    /// <summary>
+    /// Files to be ignored in AppDir
+    /// </summary>
+    public ITaskItem[]? ExcludeFromAppDir { get; set; }
+
+    /// <summary>
+    /// Path to *.app bundle
+    /// </summary>
+    [Output]
+    public string AppBundlePath { get; set; } = ""!;
+
+    /// <summary>
+    /// Path to xcode project
+    /// </summary>
+    [Output]
+    public string XcodeProjectPath { get; set; } = ""!;
+
+    public override bool Execute()
+    {
+        Utils.Logger = Log;
+        bool isDevice = Arch.Equals("arm64", StringComparison.InvariantCultureIgnoreCase);
+        if (isDevice && string.IsNullOrEmpty(CrossCompiler))
+        {
+            throw new ArgumentException("arm64 arch requires CrossCompiler");
+        }
+
+        if (!File.Exists(Path.Combine(AppDir, MainLibraryFileName)))
+        {
+            throw new ArgumentException($"MainLibraryFileName='{MainLibraryFileName}' was not found in AppDir='{AppDir}'");
+        }
+
+        // escape spaces
+        ProjectName = ProjectName.Replace(" ", "-");
+
+        string[] excludes = new string[0];
+        if (ExcludeFromAppDir != null)
+        {
+            excludes = ExcludeFromAppDir
+                .Where(i => !string.IsNullOrEmpty(i.ItemSpec))
+                .Select(i => i.ItemSpec)
+                .ToArray();
+        }
+        string[] libsToAot = Directory.GetFiles(AppDir, "*.dll")
```

Comment:
I still think this should come as input to the task (e.g. from publish)

Reply:
the `AppDir` is basically the `dotnet publish` location since anyway we need everything in the directory anyway.
Collection of files as an input makes it harder to use via command line 🙁